--- src/Form/SettingsForm.php.orig	2025-06-23 14:00:58
+++ src/Form/SettingsForm.php	2025-06-23 14:01:25
@@ -3,6 +3,7 @@
 namespace Drupal\decoupled_preview_iframe\Form;

 use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\TypedConfigManagerInterface;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Entity\EntityTypeRepositoryInterface;
 use Drupal\Core\Extension\ModuleHandlerInterface;
@@ -41,6 +42,8 @@
    *
    * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
    *   Defines the configuration object factory.
+   * @param \Drupal\Core\Config\TypedConfigManagerInterface $typed_config_manager
+   *   The typed config manager.
    * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
    *   The entity type manager service.
    * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
@@ -50,11 +53,12 @@
    */
   public function __construct(
     ConfigFactoryInterface $config_factory,
+    TypedConfigManagerInterface $typed_config_manager,
     EntityTypeManagerInterface $entity_type_manager,
     ModuleHandlerInterface $module_handler,
     EntityTypeRepositoryInterface $entity_type_repository,
   ) {
-    parent::__construct($config_factory);
+    parent::__construct($config_factory, $typed_config_manager);
     $this->entityTypeManager = $entity_type_manager;
     $this->moduleHandler = $module_handler;
     $this->entityTypeRepository = $entity_type_repository;
@@ -66,6 +70,7 @@
   public static function create(ContainerInterface $container) {
     return new static(
       $container->get('config.factory'),
+      $container->get('config.typed'),
       $container->get('entity_type.manager'),
       $container->get('module_handler'),
       $container->get('entity_type.repository'),
